### PR TITLE
Allow cell type marker gene file added to archive

### PR DESCRIPTION
**Pull request by lingyun1010**
The {marker_gene_[numbers]} are already included in the archive, and this story is to add the files named like `maker_gene_inferred_cell_type_-_ontology_labels`, so I add a new matcher.

### CODE REVIEW

**Review by alfonsomunozpomer — CHANGES_REQUESTED** [review]
Please add some tests.

File: src/test/java/uk/ac/ebi/atlas/resource/DataFileHubIT.java [hunk 1]
```diff
@@ -155,6 +156,17 @@ void findsMarkerGeneFiles() {
             assertAtlasResourceExists(subject.getSingleCellExperimentFiles(experimentAccession).markerGeneTsvs.values());
         }
 
+        @Test
+        void findsCellTypeMarkerGeneFiles(@Value("${data.files.location}") String dataFilesLocation) {
+            String experimentAccession = "E-MTAB-5061";
```
Comment (upendrakumbham) [inline src/test/java/uk/ac/ebi/atlas/resource/DataFileHubIT.java:161]: Can you make this test case a generic one? We can call `jdbcTestUtils.fetchRandomSingleCellExperimentAccessionWithMarkerGenes()` to get an experiment accession randomly.

**Review by upendrakumbham — CHANGES_REQUESTED** [review]
Please modify the test case to test cell type marker gene files randomly. If you have a dependency on the latest test data to make this test case generic, let me know I will approve this PR.

File: src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java [hunk 2]
```diff
@@ -465,6 +465,29 @@ public SingleCellExperimentFiles getSingleCellExperimentFiles(String experimentA
...
+                    if (matcher.matches()) {
+                        stringValues.add(matcher.group(1));
+                    }
+                }
+
+            } catch (IOException e) {
+                // log warning, the set will be empty, the caller decides what to do
+            }
+            return stringValues.build();
+        }
```
Comment (haideriqbal) [inline src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java:490]: Please refactor this function using `var` as variable types.

File: src/test/java/uk/ac/ebi/atlas/resource/DataFileHubIT.java [hunk 3]
```diff
@@ -155,6 +156,18 @@ void findsMarkerGeneFiles() {
...
+            String experimentAccession = jdbcUtils.fetchRandomExperimentAccession();
+            DataFileHub subject = new DataFileHub(dataFilesPath.resolve("scxa"));
+            LOGGER.info("Test cell type marker gene files for experiment {}", experimentAccession);
+            assertThat(subject.getSingleCellExperimentFiles(experimentAccession).markerGeneTsvs.values()
+                    .stream().map(path -> path.getPath()))
+                    .contains(Path.of(dataFilesLocation +
+                            "/scxa/magetab/" + experimentAccession + "/" + experimentAccession +
+                            ".marker_genes_inferred_cell_type_-_ontology_labels.tsv"));
+        }
+
```
Comment (haideriqbal) [inline src/test/java/uk/ac/ebi/atlas/resource/DataFileHubIT.java:170]: Same goes for here as mentioned above 🆙

**Review by haideriqbal — APPROVED** [review]
Please see the comments and make the changes before merging the PR

**Review by upendrakumbham — APPROVED** [review]
LGTM. Please merge away.

**Pull request merged**